Tasks/Maven/maventask.ts

```diff
@@ -282,9 +306,15 @@ function applyPmdGoals(mvnRun: trm.ToolRunner):void {
     pmd.applyPmdArgs(mvnb);
 }
 
-// Returns true if any one code analysis tool was enabled.
-function isCodeAnalysisUploadRequired():boolean {
-    return tl.getInput('pmdAnalysisEnabled', true) == 'true';
+// Returns the names of any enabled code analysis tools, or empty array if none.
+function getEnabledCodeAnalysisTools():string[] {
+    var result:string[] = [];
+
+    if (tl.getInput('pmdAnalysisEnabled', true) == 'true') {
+        result.push(pmd.toolName);
```

Comment:
All tool names should be provided by the relevant external module. This is to reduce the fragility of the code.